proposals/lip-0058.md

```diff
@@ -756,9 +730,9 @@ This function returns the slot number corresponding to the input timestamp.
 
 ```python
 def getSlotNumber(timestamp):
-    if timestamp < genesisDataStore.timestamp:
+    if timestamp < genesisTimestamp:
         raise Exception('Input timestamp cannot be smaller than genesis block timestamp.')
-    elapsedTime = timestamp - genesisDataStore.timestamp
+    elapsedTime = timestamp - genesisTimestamp
```

Comment:
oh, that's something I realized just now after having a look at LIP 0063 again. The timestamp of the genesis block could be any timepoint in its time slot. See [here](https://github.com/LiskHQ/lips/blob/ce9b653f7210f0d997a599486243ec0d031ed30b/proposals/lip-0063.md?plain=1#L332). Therefore, this calculation is not necessarily doing what it intended here. We could fix this here or in LIP 0063. I'm open for both.

For `getGeneratorAtTimestamp` I'm not sure if this is an issue. I guess not.